static-assets/components/cstudio-common/common-api.js

```diff
@@ -1871,18 +1871,18 @@ var nodeOpen = false,
                 }
               });
 
-              $modal.appendTo(window.top.document.body);
+              $modal.appendTo(getTopLegacyWindow().document.body);
```

Comment:
Try to make a single call to getTopLegacyWindow per function - it can become a little expensive to get it multiple times per function now that is not a simple property accessor.